src/foundry/foundry.js/hooks.d.ts

```diff
@@ -1,830 +1,804 @@
+     * @remarks The name for this hook is dynamically created by joining 'delete' with the type name of the Document.
      * @remarks This is called by {@link Hooks.callAll}.
-     * @see {@link SceneControls#_getControlButtons}
+     * @see {@link ClientDatabaseBackend#_postDeleteDocumentCallbacks}
+     * @see {@link TokenDocument#_onUpdateTokenActor}
      */
-    getSceneControlButtons: (controls: SceneControl[]) => unknown;
+    type DeleteDocument<D extends DocumentConstructor = DocumentConstructor> = (
+      document: D,
+      options: DocumentModificationOptions,
+      userId: string
+    ) => unknown;
 
     /**
-     * This is called after getting the {@link ContextMenu} options for the {@link SceneNavigation}, but before creating
-     * the ContextMenu.
-     * @param jq             - the JQuery of the ContextMenu parent element
-     * @param contextOptions - the already created ContextMenu.Items
-     * @remarks This is called by {@link Hooks.callAll}.
-     * @see {@link SceneNavigation#activateListeners}
+     * A hook event that fires whenever this Application is first rendered to add buttons to its header.
+     * @param app     - The Application instance being rendered
+     * @param buttons - The array of header buttons which will be displayed
+     * @typeParam A   - the type of the Application
+     * @remarks The name for this hook is dynamically created by joining 'get' with the type name of the Application and
+     * 'HeaderButtons'.
+     * @remarks This is called by {@link Hooks.call}.
+     * @see {@link Application#_getHeaderButtons}
      */
-    getSceneNavigationContext: (jq: JQuery, contextOptions: ContextMenu.Item[]) => unknown;
+    type GetApplicationHeaderButtons<A extends Application = Application> = (
+      app: A,
+      buttons: Application.HeaderButton[]
+    ) => boolean | void;
 
     /**
-     * This is called after getting the {@link ContextMenu} options for a {@link PlayerList} user, but before creating
+     * @remarks This is called after getting the {@link ContextMenu} options for a {@link ChatLog}, but before creating
      * the ContextMenu.
-     * @param jq             - the JQuery of the ContextMenu parent element
-     * @param contextOptions - the already created ContextMenu.Items
-     * @remarks This is called by {@link Hooks.callAll}.
-     * @see {@link PlayerList#activateListeners}
+     * @param jq           - the JQuery of the ContextMenu parent element
+     * @param entryOptions - the already created ContextMenuOptions
+     * @remarks The name for this hook is dynamically created by joining 'get' with the type name of the ChatLog and
+     * 'EntryContext'.
+     * @remarks This is called by {@link Hooks.call}.
+     * @see {@link ChatLog#_contextMenu}
      */
-    getUserContextOptions: (jq: JQuery, contextOptions: ContextMenu.Item[]) => unknown;
+    type GetChatLogEntryContext = (jq: JQuery, entryOptions: ContextMenu.Item[]) => boolean | void;
```

Comment:
This and the other `get...EntryContext` hook events are basically the same, they are just triggered by different sidebar tabs. Maybe it makes to group them together...? Not sure though... Might as well wait and see what foundry does about documenting them... See https://gitlab.com/foundrynet/foundryvtt/-/issues/5569